runtime/Python3/src/antlr4/xpath/XPath.py

```diff
@@ -210,24 +210,32 @@ def recover(self, e):
     def getXPathElement(self, wordToken:Token, anywhere:bool):
         if wordToken.type==Token.EOF:
             raise Exception("Missing path element at end of path")
-        word = wordToken.text
-        ttype = self.parser.getTokenType(word)
-        ruleIndex = self.parser.getRuleIndex(word)
 
+        word = wordToken.text
         if wordToken.type==XPathLexer.WILDCARD :
-
             return XPathWildcardAnywhereElement() if anywhere else XPathWildcardElement()
 
         elif wordToken.type in [XPathLexer.TOKEN_REF, XPathLexer.STRING]:
+            tsource = self.parser.getTokenStream().tokenSource
+
+            ttype = litType = None
+            if wordToken.type == XPathLexer.TOKEN_REF:
+                ttype = (tsource.ruleNames.index(word) + 1) if word in tsource.ruleNames    else None
+            else:
+                litType = tsource.literalNames.index(word)  if word in tsource.literalNames else None
+
+            # Decide which one is it
+            ttype = ttype or litType
```

Comment:
mmm... this is javascript style not python, please keep the intent explicit
also in Python 1 variable is probably sufficient